### PR TITLE
CORE-6723 add additional logging to flow event pipeline

### PR DESCRIPTION
**Pull request by LWogan**
add additional debug logging of services which are common or system. log kryo errors when they are thrown. Adjust timeout for the fiber to be less than the processor timeout so that the flow can return before it times out from the subscription thread

### CODE REVIEW

File: components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/impl/FlowEventProcessorImpl.kt [hunk 1]
```diff
@@ -55,11 +55,14 @@ class FlowEventProcessorImpl(
             return flowEventExceptionProcessor.process(t)
         }
 
+        //flow result timeout must be lower than the processor timeout as the processor thread will be killed by the subscription consumer
+        // thread after this period and so this timeout would never be reached and given a chance to return otherwise.
+        val flowTimeout = (config.getLong(PROCESSING_MAX_FLOW_EXECUTION_DURATION) * 0.75).toLong()
```
Comment (simon-johnson-r3) [inline components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/impl/FlowEventProcessorImpl.kt]: I'm not sure this is the best change to make. PROCESSING_MAX_FLOW_EXECUTION_DURATION was added by me for this purpose specifically. My misunderstanding was that a timeout in the other thread was relating to Kafka polling only, and not killing anything.

To fix this either you could make this a factor of the processor timeout and delete this value from the configuration, or you should change the default value to be lower than the other, and then document that must always be the case. If the latter, then it's probably best to run some check that these values do not collide at runtime.

**Review by simon-johnson-r3 — CHANGES_REQUESTED** [review]

File: components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/impl/FlowEventProcessorImpl.kt (hunk 1 above)
Comment (LWogan) [reply components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/impl/FlowEventProcessorImpl.kt]: I hadn't even noticed that it was a new timeout added. I thought this was pointing to PROCESSOR_TIMEOUT. ill update

File: components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/impl/FlowEventProcessorImpl.kt [hunk 2]
```diff
@@ -55,11 +55,14 @@ class FlowEventProcessorImpl(
             return flowEventExceptionProcessor.process(t)
         }
 
+        //flow result timeout must be lower than the processor timeout as the processor thread will be killed by the subscription consumer
```
Comment (simon-johnson-r3) [inline components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/impl/FlowEventProcessorImpl.kt]: very minor, but are you missing a space here?

**Review by simon-johnson-r3 — DISMISSED** [review]

**Comment by simon-johnson-r3** [comment]
I assume you'll raise an api PR at some point after this is merged to remove the value there?

**Comment by LWogan** [comment]
> I assume you'll raise an api PR at some point after this is merged to remove the value there?

https://github.com/corda/corda-api/pull/600 got one here. its already plugged into this pr to make sure i removed all references to the path, of which there was a few i did miss

**Review by simon-johnson-r3 — APPROVED** [review]

**Review by JamesHR3 — APPROVED** [review]

**Pull request merged**